pkg/operator/ceph/cluster/monitoring.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+)
+
+func (c *ClusterController) ConfigureCephMonitoring(cluster *cluster, cephUser string) {
+	var isEnabled bool
+	daemons := []string{"mon", "osd", "status"}
+
+	for _, daemon := range daemons {
+		// Is the monitoring enabled for that daemon?
+		isEnabled = isMonitoringConfigured(daemon, cluster.Spec)
+		if health, ok := cluster.monitoringChannels[daemon]; ok {
+			if health.monitoringRunning {
+				// If the goroutine was running but the CR was updated to stop the monitoring we need to close the channel
+				if isEnabled {
```

Comment:
This seemed backwards to me. You only want to stop the monitoring if it's not enabled, right?
```suggestion
				if !isEnabled {
```

Reply:
Th logic is backward anyway since we use "disabled" in the CRD. Will try to make it more readeable.